cglib/src/test/java/net/sf/cglib/proxy/TestEnhancer.java

```diff
@@ -243,6 +246,58 @@ public void testCustomClassLoader()throws Throwable{
         
     }
 
+    /**
+     * Verifies that the cache in {@link AbstractClassGenerator} SOURCE doesn't
+     * leak class definitions of classloaders that are no longer used.
+     * <p/>
+     * Without the proper fix, this test throws {@link OutOfMemoryError}:
+     * PermGen
+     * <p/>
+     * This test takes around 30 secs.
```

Comment:
It can be faster if you just test that a single classloader gets unloaded via `PhantomReference<ClassLoader> -> ReferenceQueue`.
No need to wait for OOM, you just need to test that a single classloader can be unloaded.